### DISCUSSION TITLE
Is there a plan to support NoSQL metadata sources like mongodb, elasticsearch, and redis?

### DISCUSSION THREAD

**Discussion by beryllw** [apache/gravitino#5499] [Q&A]
Now gravitino supports Schema, Files, Model, Topics catalog.Is there a plan to support NoSQL metadata sources?

**Comment by mchades** [comment]
Good point, but can you provide some specific scenarios where NoSQL needs to manage metadata?

**Reply by beryllw** [reply]
If users manage the Schema data source with Gravitino, why don't they use Gravitino to manage the NoSQL data source as well?
Gravitiono can provide permission control and data source management for NOSQL datasource.

When I use Spark to sync data from MySQL to Elasticsearch, Gravitino can offer similar functionalities for Elasticsearch as it does for MySQL, such as permission control and metadata management.

**Reply by czxxing** [reply]
we have the same requirements，including elasticsearch、mongo and redis；

**Reply by adwaithadwaith** [reply]
@czxxing Did you find any way to integrate mongo ?

**Comment by justinmclean** [comment]
Note also that Elasticsearch, Redis and MongoDB are not available under an open-source license. Depending on how they need to be implemented, this might cause issues when integrating them with Apache Gravitino as their licenses are incompatible with the Apache license.
